src/components/MultiSelect/MultiSelect.js

```diff
@@ -119,7 +113,19 @@ export default class MultiSelect extends React.Component {
       // Reference: https://github.com/paypal/downshift/issues/206
       case Downshift.stateChangeTypes.clickButton:
       case Downshift.stateChangeTypes.keyDownSpaceButton:
-        this.handleOnToggleMenu();
+        this.setState(() => {
+          let nextIsOpen = changes.isOpen;
+          if (changes.isOpen === false) {
```

Comment:
@rezak-otmani instead of doing multiple if-statements, you can combine them into 
```
if (changes.isOpen === false && this.inputNode === document.activeElement) {
  nextIsOpen = true;
}
```

Reply:
Hi @werne2j  thank's for your review, preference to leave the structure of the code as it is, and keep the comments as they are important so that this piece of code be clear.

Reply:
While I see there are some cases that nested `if` approach makes the code more clear, I’d avoid using it as the general pattern.